src/graph.rs

```diff
@@ -39,6 +43,62 @@ impl<'a, N: Clone, E: Clone, Ty: EdgeType> Graph<N, E, Ty> {
         })
     }
 
+    pub fn edge_by_screen_pos(
+        &self,
+        meta: &'a Metadata,
+        style: &'a SettingsStyle,
+        screen_pos: Pos2,
+    ) -> Option<(EdgeIndex, &Edge<E>)> {
+        let pos_in_graph = (screen_pos.to_vec2() - meta.pan) / meta.zoom;
+        self.edges_iter().find(|(idx, e)| {
+            let (source, target) = self.g.edge_endpoints(*idx).unwrap();
+            let source_pos = self.g.index(source).location();
+            let target_pos = self.g.index(target).location();
+            if source == target { // edge is a loop (bezier curve)
+                // TODO lots of duplicated code (only slightly changed from draw/edge
+                let node = self.g.index(source);
+                let rad = node.screen_radius(meta, style) / meta.zoom;
+                let center = source_pos;
+                let center_horizon_angle = PI / 4.;
+                let y_intersect = center.y - rad * center_horizon_angle.sin();
+
+                let edge_start = Pos2::new(center.x - rad * center_horizon_angle.cos(), y_intersect);
+                let edge_end = Pos2::new(center.x + rad * center_horizon_angle.cos(), y_intersect);
+
+                // TODO what does `order` do?
```

Comment:
Order is basically an index of duplicate edges, because it can be more than one. In edge drawing function I use order to increase loop radius for lopped edge or to increase curvature for duplicate edges between 2 different nodes.